test/e2e/framework/framework.go

```diff
@@ -154,14 +154,12 @@ func (f *Framework) AfterEach() {
 	defer f.DestroyEnvironment()
 
 	defer func(kubeClient kubernetes.Interface, ingressclass string) {
-		go func() {
-			defer ginkgo.GinkgoRecover()
-			err := deleteIngressClass(kubeClient, ingressclass)
-			assert.Nil(ginkgo.GinkgoT(), err, "deleting IngressClass")
-		}()
+		defer ginkgo.GinkgoRecover()
+		err := deleteIngressClass(kubeClient, ingressclass)
+		assert.Nil(ginkgo.GinkgoT(), err, "deleting IngressClass")
```

Comment:
I suggest not to modify here. 
Here we should let it be handled in a closure, due to this is `AfterEach`,  IIRC

Reply:
IIRC, once the `go func` is still here, it will not pass the test. Let me try.

Reply:
I got it wrong, and changed it back to the original case.